guides/plugins/plugins/checkout/cart/tax-provider.md

```diff
@@ -0,0 +1,213 @@
+        );
+
+        $id = Uuid::randomBytes();
+
+        $connection->insert(TaxProviderDefinition::ENTITY_NAME, [
+            'id' => $id,
+            'identifier' => Swag\BasicExample\Checkout\Cart\Tax\TaxProvider::class, // your tax provider class
+            'active' => true,
+            'priority' => 1,
+            'availability_rule_id' => $ruleId,
+            'created_at' => (new \DateTime())->format(Defaults::STORAGE_DATE_TIME_FORMAT),
+        ]);
+    }
+
+    public function updateDestructive(Connection $connection): void
+    {
+    }
+}
+```
+
+{% endcode %}
+
+### Via repository
+
+If you don't want to use migrations, you can also create your tax provider via repository. A good place for doing so would be the `install` lifecycle method of your plugin.
+
+If you do not know of plugin lifecycle methods yet, please refer to our [plugin lifecycle guide](../../plugin-fundamentals/plugin-lifecycle.md).
+
+{% code title="<plugin root>/src/BasicExample.php" %}
+
+```php
+<?php declare(strict_types=1);
+
+namespace SwagBasicExample;
+
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\EqualsFilter;
+use Shopware\Core\Framework\Plugin;
+use Shopware\Core\Framework\Plugin\Context\InstallContext;
+use Shopware\Core\Framework\Uuid\Uuid;
+
+class SwagBasicExample extends Plugin
+{
+    public function install(InstallContext $installContext): void
+    {
+        $ruleRepo = $this->container->get('rule.repository');
+
+        $criteria = new Criteria();
+        $criteria->addFilter(new EqualsFilter('name', 'Always valid (Default)'));
+
+        $rule = $ruleRepo->search($criteria, $installContext->getContext())->first();
+
+        $taxRepo = $this->container->get('tax_provider.repository');
+        $taxRepo->create([
+            [
+                'id' => Uuid::randomHex(),
+                'identifier' => Swag\BasicExample\Checkout\Cart\Tax\TaxProvider::class,
+                'priority' => 1,
+                'active' => false, // activate this via the `activate` lifecycle method
+                'availabilityRule' => $rule,
+                'translations' => [
+                    'de-DE' => [
+                        'name' => 'TaxJar',
+                    ],
+                    'en-GB' => [
+                        'name' => 'TaxJar',
+                    ],
+                ],
+            ],
+        ], $installContext->getContext());
+    }
+    
+    // do not forget to add an `uninstall` method, which removes your tax providers, when the plugin is uninstalled
+    // you also may want to add the `activate` and `deactivate` methods to activate and deactivate the tax providers
+}
+
+```
+
+{% endcode %}
+
+If you have taken these steps, your tax provider should be ready and already used during checkout. You should also see the new tax provider showing up in the administration in `Settings > Tax`.
```

Comment:
I'd like to see that hint in the instructions for the app as well. They wouldn't know where to see and configure these without looking here.